Classes/View/Slot/Service/SlotFlexFormService.php

```diff
@@ -0,0 +1,206 @@
+<?php
+
+/*
+ * Copyright (C) 2017
+ * Nathan Boiron <nathan.boiron@gmail.com>
+ * Romain Canon <romain.hydrocanon@gmail.com>
+ *
+ * This file is part of the TYPO3 NotiZ project.
+ * It is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License, either
+ * version 3 of the License, or any later version.
+ *
+ * For the full copyright and license information, see:
+ * http://www.gnu.org/licenses/gpl-3.0.html
+ */
+
+namespace CuyZ\Notiz\View\Slot\Service;
+
+use CuyZ\Notiz\Definition\Tree\EventGroup\Event\EventDefinition;
+use CuyZ\Notiz\Service\CacheService;
+use CuyZ\Notiz\View\Slot\Application\Slot;
+use CuyZ\Notiz\View\Slot\SlotView;
+use CuyZ\Notiz\View\ViewPathsAware;
+use TYPO3\CMS\Core\SingletonInterface;
+
+class SlotFlexFormService implements SingletonInterface
+{
+    /**
+     * @var SlotViewService
+     */
+    protected $slotViewService;
+
+    /**
+     * @var CacheService
+     */
+    protected $cacheService;
+
+    /**
+     * @param SlotViewService $slotViewService
+     * @param CacheService $cacheService
+     */
+    public function __construct(SlotViewService $slotViewService, CacheService $cacheService)
+    {
+        $this->slotViewService = $slotViewService;
+        $this->cacheService = $cacheService;
+    }
+
+    /**
+     * Builds a list of FlexForms, based on the slots of every event.
+     *
+     * Events with no slots are ignored.
+     *
+     * @param ViewPathsAware $viewPaths
+     * @return array
+     */
+    public function getNotificationFlexFormList(ViewPathsAware $viewPaths)
+    {
+        $flexFormList = ['default' => $this->getDefaultFlexForm()];
+
+        foreach ($this->slotViewService->getEventsWithSlots($viewPaths) as $event => $view) {
+            $flexForm = $this->getSlotViewFlexForm($view);
+
+            /** @var EventDefinition $event */
+            $flexFormList[$event->getFullIdentifier()] = $flexForm;
+        }
+
+        return $flexFormList;
+    }
+
+    /**
+     * @param SlotView $view
+     * @return string
+     */
+    public function getSlotViewFlexForm(SlotView $view)
+    {
+        $hash = 'slots-' . sha1(serialize([
+                $view->getTemplatePathAndFilename(),
+                $view->getLayoutRootPaths(),
+                $view->getPartialRootPaths(),
+                $view->getEventDefinition()->getFullIdentifier()
+            ]));
```

Comment:
Do you think that it would be better to have that in a `getHash` method in `SlotView`?